analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd

```diff
@@ -0,0 +1,591 @@
+  #
+  # Args:
+  #   tmb_df: a chromosomal breaks density file path where each sample is
+  #                 a row with `samples` and `breaks_count` columns. 
+  #   plot_title: to be used for the ggplot2::ggtitle and what it will be saved 
+  #                as a png as.
+  
+  cdf_plot <- tmb_df %>%
+    as.data.frame() %>%
+    dplyr::mutate(short_histology = tools::toTitleCase(short_histology)) %>%
+    # Only plot histologies groups with more than `min_samples` number of samples
+    dplyr::group_by(short_histology, add = TRUE) %>%
+    # Only keep groups with this amount of samples
+    dplyr::filter(dplyr::n() > 5) %>%
+    # Calculate histology group mean
+    dplyr::mutate(
+      hist_mean = mean(tmb),
+      hist_rank = rank(tmb, ties.method = "first") / dplyr::n(),
+      sample_size = paste0("n = ", dplyr::n())
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(short_histology = reorder(short_histology, hist_mean)) %>%
+    # Now we will plot these as cummulative distribution plots
+    ggplot2::ggplot(ggplot2::aes(
+      x = hist_rank,
+      y = tmb,
+      color = datasets
+    )) +
+    ggplot2::geom_point() +
+    # Add summary line for mean
+    ggplot2::geom_segment(
+      x = 0, xend = 1, color = "grey",
+      ggplot2::aes(y = hist_mean, yend = hist_mean)
+    ) +
+    # Separate by histology
+    ggplot2::facet_wrap(~ short_histology + sample_size, nrow = 1, strip.position = "bottom") +
+    ggplot2::theme_classic() +
+    ggplot2::xlab("") +
+    ggplot2::ylab("TMB") +
+    # Transform to log10 make non-log y-axis labels
+    ggplot2::scale_y_continuous(trans = "log1p", breaks = c(0, 1, 3, 10, 30)) +
+    ggplot2::scale_x_continuous(limits = c(-0.2, 1.2), breaks = c()) +
+    # Making it pretty
+    ggplot2::theme(legend.position = "none") +
+    ggplot2::theme(
+      axis.text.x = ggplot2::element_blank(),
+      axis.ticks.x = ggplot2::element_blank(),
+      strip.placement = "outside",
+      strip.text = ggplot2::element_text(size = 10, angle = 90, hjust = 1),
+      strip.background = ggplot2::element_rect(fill = NA, color = NA)
+    ) +
+    ggplot2::ggtitle(plot_title)
+
+  # Save as a PNG
+  ggplot2::ggsave(filename = file.path(plots_dir, paste0("tmb-cdf-", plot_title, ".png")), 
+                  plot = cdf_plot, 
+                  width = 10, 
+                  height = 7.5)
+}
+```
+
+Function for comparing the sequence overlap of two GenomicRanges objects. 
+
+```{r}
+bed_overlap <- function(bed_granges_1,
+                        bed_granges_2,
+                        name_1,
+                        name_2,
+                        plot_name) {
+  # Given two GenomicRanges objects make a VennDiagram of their overlap
+  
+  # Find intersection
+  overlaps <- GenomicRanges::intersect(bed_granges_1, bed_granges_2)
+
+  # Percent of TCGA Target region covered by overlap
+  sum(overlaps@ranges@width) / sum(bed_granges_2@ranges@width)
```

Comment:
Do we need to collapse (reduce?) these before summing? I see that you do it later, but I  feel like the more bombproof way do to it is in this function.